src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,400 @@
+
+## What is a REST API?
+REST is a framework that outlines a collection of guidelines for building web services. A REST API offers a straightforward and adaptable method to interact with web services without the need for any processing. REST is often favored for its lower bandwidth usage, simplicity, and flexibility, making it better suited for internet applications. REST is utilized to retrieve or transmit data from a web service with all interactions through a REST API occurring via HTTP requests.
+
+### Key Principles of REST
+The four key principles of REST API design, derived from the broader principles of REST architecture, are:
+
+**1. Stateless**: Every client request to the server must include all necessary information for comprehension and processing. The server should not retain any data about the client's latest HTTP request.
+
+**2. Client-Server Architecture**: The client handles the user interface and experience, while the server manages request processing and resource management.
+
+**3. Cacheable**: Clients can cache responses from the server, enhancing performance by reducing the server load and minimizing client-server interactions.
+
+**4. Uniform Interface**: The API has a consistent interface, which simplifies the architecture and enhances visibility of interactions.
+
+## REST API Architecture
+REST APIs utilize a client-server architecture, communicating over HTTP. The client sends requests to the server, which processes them and sends back responses. Communication is stateless, requiring each client request to contain all necessary information for processing.
+
+![REST API Archiecture](/src/assets/RestApi/images/rest_api.png)
+
+### Components of REST API Architecture
+- **Resources**: In REST, resources are identified by URIs, representing anything that can be named and addressed, such as documents, images, temporal services, collections of other resources, non-virtual objects, and more.
+
+- **HTTP Methods**: REST APIs utilize standard HTTP methods for operations on resources. Common methods include GET (to retrieve a resource), POST (to create a new resource), PUT (to update a resource), and DELETE (to remove a resource).
+
+- **Representations**: Resources are represented in a format easily understood by both client and server, commonly JSON (JavaScript Object Notation) or XML (eXtensible Markup Language).
+
+## Understanding Endpoints in REST API
+Endpoints are the points of interaction between the client and the server in a RESTful API. Each endpoint is associated with a specific URI where an API can be accessed, and it defines a specific URI at which the resource can be accessed or manipulated. Endpoints are crucial for REST APIs as they provide a structured and standardized way for clients to interact with the API's resources.
```

Comment:
The sentence above is a bit confusing to me. You write above "...where an API can be accessed". Since you're talking about a specific URI, I guess it's a specific function or method of the API that you're referring to?

Reply:
I might have made mistake while paraphrasing. Updated according to https://www.sensedia.com/pillar/components-of-an-api-endpoint-resources-methods#:~:text=An%20endpoint%20is%20a%20specific,resource%20offered%20by%20the%20API.